src/imio/helpers/date.py

```diff
@@ -0,0 +1,164 @@
+
+def wordizeDate(date, context=None, long_format=False):
+    """
+      Convert a date made of digits to a date made of words.
+      If long_format is True, we convert the hour to words too
+    """
+    converted_month = formatDate(date, month_name=True, context=context, long_format=False).split(' ')[1]
+
+    splitted_date = date.strftime("%Y/%m/%d").split('/')
+    date_time = ""
+    #check if we have a time
+    if len(str(date).split(' ')) > 1:
+        date_time = str(date).split(' ')[1]
+
+    #work on the date
+    day_to_convert = int(splitted_date[2])
+    if day_to_convert == 1:
+        converted_day = "premier"
+    else:
+        converted_day = int2word(day_to_convert)
+    converted_year = int2word(splitted_date[0])
+    converted_date = "%s %s %s" % (converted_day, converted_month, converted_year)
+
+    converted_time = ""
+    if long_format and date_time:
+        #work on the hour
+        splitted_time = date_time.split(':')
+        hour = splitted_time[0]
+        minutes = splitted_time[1]
+        converted_hour = int2word(int(hour))
+        if converted_hour.endswith("un "):
+            converted_hour = converted_hour[:-1] + "e"
+        converted_minutes = int2word(int(minutes))
+        if hour in ['01', '1', '0', '00', ]:
+            hour_word = "heure"
+        else:
+            hour_word = "heures"
+        #do not display 'zero' in the date like : eleven hour zero...
+        #see http://dev.communesplone.org/trac/ticket/1553
+        if converted_minutes and not minutes in ['0', '00',]:
+            converted_time = " %s %s %s %s" % (u"à", converted_hour, hour_word, converted_minutes)
+        else:
+            #remove the minutes, especially useless spaces if there are no minutes
+            converted_time = " %s %s %s" % (u"à", converted_hour, hour_word)
+    return converted_date.encode('utf-8') + converted_time.encode('utf-8')
+
+def int2word(n):
+    """
+      Convert an integer number into a string of french word
+      Based on the http://www.daniweb.com/code/snippet609.html
+      code that convert into english words...
+    """
+    ############# globals ################
+    ones = ["", "un ","deux ","trois ","quatre ", "cinq ",
+    "six ","sept ","huit ","neuf "]
+    tens = ["dix ","onze ","douze ","treize ", "quatorze ",
+    "quinze ","seize ","dix-sept ","dix-huit ","dix-neuf "]
+    twenties = ["","","vingt ","trente ","quarante ",
+    "cinquante ","soixante ","septante ","quatre-vingt ","nonante "]
+    thousands = ["","mille ","million ", "milliard ", "billiard ",
+    "trilliard ", "quadrilliard ", "quintillard ", "sextilliard", "septilliard ",
+    "octilliard ", "nonilliard ", u"dÃ©cillion ", u"dÃ©cilliard ", u"unodÃ©cillion ", u"unodÃ©cilliard ", u"duodÃ©cillion ", u"duodÃ©cilliard "]
+
+    #quickly manage if n is < 10
+    if n < 10:
+        ones = [u'zéro', "un ","deux ","trois ","quatre ", "cinq ", "six ","sept ","huit ","neuf "]
```

Comment:
Space after ',', is a blank after u'zéro' not needed?  What about something like 0h01?